sql/core/src/test/scala/org/apache/spark/sql/DateFunctionsSuite.scala

```diff
@@ -475,6 +1164,45 @@ class DateFunctionsSuite extends QueryTest with SharedSQLContext {
       Row(ts1.getTime / 1000L), Row(ts2.getTime / 1000L)))
   }
 
+  test("to_unix_timestamp with session local timezone") {
```

Comment:
These newly added tests are so similar that they all try to prove one thing: when you convert string or date to timestamp, the result changes according to session local timezone. When you convert timestamp to string or date, the result also changes with session local timezone. All time-related expressions should respect this.

shall we just write a general test for this? then we don't need so many similar tests